src/language-server/diagnostics.ts

```diff
@@ -117,18 +121,28 @@ export default class DiagnosticGenerator extends Handler {
   private updateLinter() {
     let rules: Iterable<Rule> = new Set();
     const projectConfig = this.settings.projectConfig;
-    if (projectConfig.lint && projectConfig.lint.rules) {
-      try {
-        rules = registry.getRules(projectConfig.lint.rules);
-      } catch (e) {
-        // TODO(rictic): let the user know about this error, and about
-        //   this.settings.projectConfigDiagnostic if it exists.
+    if (projectConfig.lint) {
+      const lintConfig = projectConfig.lint;
+      if (lintConfig.rules) {
+        try {
+          rules = registry.getRules(projectConfig.lint.rules);
+        } catch (e) {
+          // TODO(rictic): let the user know about this error, and about
+          //   this.settings.projectConfigDiagnostic if it exists.
+        }
+      }
+      if (lintConfig.ignoreWarnings) {
+        this.warningCodesToFilterOut =
+            new Set(projectConfig.lint.ignoreWarnings);
```

Comment:
nit: use the shorter `lintConfig.ignoreWarnings`. You could avoid the `if/else` check and just set 
```ts
this.warningCodesToFilterOut = new Set(lintConfig.ignoreWarnings);